fed/_private/global_context.py

```diff
@@ -28,27 +35,40 @@ def next_seq_id(self) -> int:
     def get_cleanup_manager(self) -> CleanupManager:
         return self._cleanup_manager
 
-    def job_name(self) -> str:
+    def get_job_name(self) -> str:
         return self._job_name
 
+    def get_failure_handler(self) -> Callable[[], None]:
+        return self._failure_handler
+
+    def acquire_shutdown_flag(self) -> bool:
+        with threading.Lock():
+            if self._atomic_shutdown_flag:
```

Comment:
We should add a detail comment to tell what these lines do because the implementation is too navie.